rviz_default_plugins/src/rviz_default_plugins/displays/map/map_display.cpp

```diff
@@ -547,10 +547,11 @@ void MapDisplay::transformMap()
 
   Ogre::Vector3 position;
   Ogre::Quaternion orientation;
-  if (!context_->getFrameManager()->transform(frame_, transform_time, current_map_.info.origin,
-    position, orientation) &&
-    !context_->getFrameManager()->transform(frame_, context_->getClock()->now(),
-    current_map_.info.origin, position, orientation))
+  if (!context_->getFrameManager()->transform(
+      frame_, transform_time, current_map_.info.origin, position, orientation) &&
+    !context_->getFrameManager()->transform(
+      frame_, rclcpp::Time(0, 0, context_->getClock()->get_clock_type()),
```

Comment:
Previously, the code was getting the current time (now) from the context's clock. The current code initializes the time to 0,0. Isn't this different, semantically? That is, getting a transform at time 0 versus now?

Reply:
It is semantically different. Providing time zero in this case means "get the latest available transform" versus getting the transform for time "now". IIRC from my investigation (https://github.com/ros2/rviz/pull/483#issue-345994593), it's possible a transform does not exist for time "now". This change let's us gracefully fallback to the best available transform.

Reply:
As an aside, I've opened up a ticket to check for similar timing "bugs" in other displays (https://github.com/ros2/rviz/issues/484).

Reply:
Ok, thanks for the info.